pkg/controller/common/license/model.go

```diff
@@ -69,7 +72,12 @@ func (l EnterpriseLicense) IsValid(instant time.Time) bool {
 
 // IsTrial returns true if this is a self-generated trial license.
 func (l EnterpriseLicense) IsTrial() bool {
-	return l.License.Type == LicenseTypeEnterpriseTrial
+	return l.License.Type == LicenseTypeEnterpriseTrial || l.License.Type == LicenseTypeLegacyTrial
+}
+
+// IsAutoGenerated returns true if this license has been generated by trial controller or is eligible to be turned into one.
+func (l EnterpriseLicense) IsAutoGenerated() bool {
```

Comment:
could we rename this to `IsTrialCandidate`?

Reply:
Hm I wanted to capture that these trials originate from inside the operator. There are externally generated trials as well the we are not interested here that are also `TrialCandidates` or maybe I am misunderstandig.

Reply:
Got it. I think `IsAutoGenerated` does not reflect the function is interested in trial licenses.
Maybe something like `IsECKManagedTrial`?

Reply:
Yes that's good!